pkg/api/handlers.go

```diff
@@ -161,6 +162,27 @@ func DefaultConfigHandler(actualCfg interface{}, defaultCfg interface{}) http.Ha
 	}
 }
 
+type configResponse struct {
+	Status string `json:"status"`
+	Config string `json:"data"`
+}
+
+func (cfg *Config) statusConfigHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		data, err := yaml.Marshal(cfg)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Does the Prometheus API return errors in a particular response? e.g. `{"status":"error", "error":"YAML problems!"}`

Reply:
No, it just writes a string error to the response. Please see https://github.com/prometheus/prometheus/blob/c3fac587ef3c7d515e319844042ee0dddab54cbc/config/config.go#L252.

Reply:
Ah, it doesn't use `"error":err` but it looks like errors will still get wrapped in something like the `configResponse` struct you created here. Can you make sure we do the same thing?

https://github.com/prometheus/prometheus/blob/c3fac587ef3c7d515e319844042ee0dddab54cbc/web/api/v1/api.go#L1336-L1341

Reply:
Oh, nevermind. I see you've already done it!